scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+def get_repository_commit_sha(repository_location):
+    command = f"git -C {repository_location} rev-parse HEAD"
+    process = subprocess.run(command.split(), check=True, stdout=subprocess.PIPE)
+    return process.stdout.decode('ascii').strip()
+
+def update_ncs(repository_location, version):
+    # Fetch sdk-nrf to the desired version.
+    command = f"git -C {repository_location} fetch origin"
+    subprocess.run(command.split(), check=True)
+
+    # Checkout sdk-nrf to the desired version.
+    command = f"git -C {repository_location} checkout {version}"
+    subprocess.run(command.split(), check=True)
+
+    # Call west update command to update all projects and submodules used by sdk-nrf.
+    command = "west update -r"
+    subprocess.run(command.split(), check=True)
+
+def get_ncs_recommended_version():
+    chip_root = os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.normpath('../../..')))
+
+    # Read recommended version saved in the .nrfconnect-recommended-version file.
+    try:
+        with open(os.path.join(chip_root, 'config/nrfconnect/.nrfconnect-recommended-version'), 'r') as f:
+            return f.readline().strip()
+    except:
+        raise RuntimeError("Encountered problem when trying to read .nrfconnect-recommended-version file.")
+
+def print_messages(messages : list, yellow_text :  bool):
+    # Add colour formatting if yellow text was set
+    if yellow_text:
+        messages = [f"\33[33m{message}\x1b[0m" for message in messages]
+
+    for message in messages:
+        print(message)
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = f"WARNING: Your current NCS version ({current_version})"
```

Comment:
You could use an array `lines = [f"WARN...` here to simplify the `print_messages` call.